upgrade-2.x.sh

```diff
@@ -623,6 +623,22 @@ function find_partitions {
     log "Update partition: ${update_part}"
 }
 
+#######################################
+# In some cases, the state partition does not have the required `resin-state` label
+# and upon reboot the device would be inoperable. Apply that label as needed.
+# Globals:
+#   root_dev
+#   state_part
+# Returns:
+#   None
+#######################################
+function fix_state_label() {
+    if [ "${state_part}" != "/mnt/state" ]; then
+        log "Need to apply proper state partition label."
+        e2label ${root_dev}5 "resin-state"
```

Comment:
Do we have boards with different partition indexes? O remember tx2 had something like that. @telphan ? @floion ?

Reply:
Yes @agherzan @imrehg 
The edison has a different layout and the 5th partition is not the state partition
And @telphan can chime in for the TX2 / TX1 (actually should chime in)

Reply:
Yes, state partition is different in TX1 and TX2, I think it is 14 and 16 respectively.

Reply:
Thanks guys.

Reply:
@agherzan I have a better way to handle this in mind, will try it out and rewrite if that works.